tasks/ensure-api-healthy/main.go

```diff
@@ -24,9 +25,20 @@ func main() {
 
 	for startTime := time.Now(); time.Since(startTime).Seconds() < MAX_TIMEOUT_IN_SEC; time.Sleep(5 * time.Second) {
 		resp, err := client.Get(url)
-		if err != nil || resp.StatusCode != 200 {
+		if err != nil {
 			numSuccessResponses = 0
-			fmt.Printf("Received error from the API, resetting...\n")
+			fmt.Printf("Received error when requesting API, resetting...\n")
+			fmt.Printf(err.Error() + "\n")
+			continue
+		}
+		if resp.StatusCode != 200 {
+			numSuccessResponses = 0
+			fmt.Printf("Received HTTP %d from the API, resetting...\n", resp.StatusCode)
+			body, err := io.ReadAll(resp.Body)
+			if err == nil {
+				fmt.Printf(string(body) + "\n")
+				resp.Body.Close()
```

Comment:
Should calling `Close()` be "defer'ed" before reading from resp.Body?


```golang
defer resp.Body.Close()
body, err := io.ReadAll(resp.Body)
if err == nil {
...
```

Reply:
I tried, but it doesn't really work because we're inside a loop. So I applied this suggested solution: https://stackoverflow.com/a/45617791

Reply:
That makes sense. In that case, should it be moved out of the error check so that it is always called before continuing?

Reply:
I think we only need to call `Close()` when we call `ReadAll(resp.Body)` AND that function did not return an error. So I would say that this is the only place for `Close()`.

Reply:
How about [using a function](https://stackoverflow.com/questions/45617758/proper-way-to-release-resources-with-defer-in-a-loop) (anonymous or named) to hold the logic in the loop? That way, each iteration could defer closing the response body before moving on to the next request.